### PR TITLE
[ci] fix ci test fused_moe op

### PR DESCRIPTION
**Pull request by BBuf**
## Motivation

I found that the errors related to "failed to run tuning_fused_moe_triton.py" in https://github.com/sgl-project/sglang/issues/4991 are due to a particular commit in the CI that mistakenly deleted test_fused_moe.py. This deletion resulted in a lack of tests to check for potential circular import issues caused by the line `from sglang.srt.layers.moe.fused_moe_triton.fused_moe import fused_moe`. The test has been added back. Additionally, I simplified the tests by removing the parts related to vllm `fused_moe` , and now it only compares with the naive implementation in torch. .cc @zhyncs 

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: test/srt/test_fused_moe.py [hunk 1]
```diff
@@ -98,22 +97,11 @@ def _test_case(self, m, n, k, e, topk, dtype, use_fp8_w8a8=False):
...
-                w2,
-                score,
-                topk,
-                renormalize=False,
-                use_fp8_w8a8=True,
-                w1_scale=w1_scale,
-                w2_scale=w2_scale,
-                a1_scale=a1_scale,
-                a2_scale=a2_scale,
+            torch_output = self.torch_naive_moe(a, w1, w2, score, topk)
```
Comment (fzyzcjy) [inline test/srt/test_fused_moe.py]: nit: a bit confused, it seems we need the w1_scale etc when doing moe, and torch_naive_moe does not have this

Comment (BBuf) [reply test/srt/test_fused_moe.py:100]: It's no necessary, please refer to https://github.com/deepseek-ai/DeepGEMM/blob/main/tests/test_core.py#L33 .

Comment (BBuf) [reply test/srt/test_fused_moe.py]: Fixed.

File: test/srt/test_fused_moe.py [hunk 2]
```diff
@@ -45,20 +44,57 @@ def get_tolerance(self, dtype):
...
         topk_weight = topk_weight.view(-1)
         topk_ids = topk_ids.view(-1)
-        for i in range(w1.shape[0]):
-            mask = topk_ids == i
-            if mask.sum():
-                out[mask] = SiluAndMul()(a[mask] @ w1[i].transpose(0, 1)) @ w2[
-                    i
-                ].transpose(0, 1)
+
+        # Handle FP8 mode
```
Comment (fzyzcjy) [inline test/srt/test_fused_moe.py]: nit: what about

```
def torch_naive_moe():
  # part 1: if it is fp8+scale, convert to fp16
  w1 = w1 * scale, etc
  # part 2: original bf16 code
  SiluAndMul, etc
```

Comment (BBuf) [reply test/srt/test_fused_moe.py]: Make sense, fixed again.

File: test/srt/test_fused_moe.py [hunk 3]
```diff
@@ -45,20 +44,49 @@ def get_tolerance(self, dtype):
...
         a = a.view(B, -1, D).repeat(1, topk, 1).reshape(-1, D)
         out = torch.zeros(B * topk, w2.shape[1], dtype=a.dtype, device=a.device)
         score = torch.softmax(score, dim=-1, dtype=torch.float32)
         topk_weight, topk_ids = torch.topk(score, topk)
         topk_weight = topk_weight.view(-1)
         topk_ids = topk_ids.view(-1)
-        for i in range(w1.shape[0]):
+
+        if w1.dtype == torch.float8_e4m3fn:
+            w1_orig = w1.to(a.dtype)
```
Comment (fzyzcjy) [inline test/srt/test_fused_moe.py]: nit: wondering whether the name can be slightly optimized

**Review by fzyzcjy — APPROVED** [review]

File: test/srt/test_fused_moe.py (hunk 3 above)
Comment (BBuf) [reply test/srt/test_fused_moe.py]: Make sense, I have renamed it to `w1/2_compute`.
